src/main/java/polyglot/Main.java

```diff
@@ -56,12 +61,19 @@ public static void main(String[] args) {
     logger.info("Creating dynamic grpc client");
     DynamicGrpcClient dynamicClient;
     if (arguments.oauthConfig().isPresent()) {
-      Credentials credentials = new RefreshTokenCredentials(
-          arguments.oauth2RefreshToken(),
-          arguments.oauthConfig().get(),
-          Clock.systemDefaultZone());
-      dynamicClient = DynamicGrpcClient.createWithCredentials(
-          methodDescriptor, arguments.endpoint(), arguments.useTls(), credentials);
+      Credentials credentials;
+      if (arguments.oauth2AccessToken().isPresent()) {
+        logger.info("Using provided access token with 1 day expiry period");
+        credentials = new OAuth2Credentials(new AccessToken(arguments.oauth2AccessToken().get(),
+                Date.from(Clock.systemDefaultZone().instant().plus(1, ChronoUnit.DAYS))));
```

Comment:
It appears to be legal [1] to pass null as an expiration time, could you pass

null /* expirationTime */

instead of the magic value here?


[1] https://github.com/google/google-auth-library-java/blob/master/oauth2_http/java/com/google/auth/oauth2/AccessToken.java#L33